### PR TITLE
Add urlopen with retry

### PR DESCRIPTION
**Pull request by teetone**
### Reasons for making this change

Added a retry mechanism for `urlopen` to help avoid failures from a single request attempt.

### Related issues

Resolves #2892 

### Checklist

* [x] I've added a screenshot of the changes, if this is a frontend change
* [ ] I've added and/or updated tests, if this is a backend change
* [x] I've run the [pre-commit.sh](https://github.com/codalab/codalab-worksheets/blob/master/pre-commit.sh) script
* [x] I've updated [docs](https://github.com/codalab/codalab-worksheets/tree/master/docs), if needed

### CODE REVIEW

**Review by epicfaace — COMMENTED** [review]
This looks good -- what do you think of the alternative of just replacing calls to urlopen to `requests` (which uses retries automatically)?

**Comment by nelson-liu** [comment]
> This looks good -- what do you think of the alternative of just replacing calls to urlopen to requests (which uses retries automatically)?

+1 for just using requests

**Comment by teetone** [comment]
> This looks good -- what do you think of the alternative of just replacing calls to urlopen to `requests` (which uses retries automatically)?

Correct me if I'm wrong, but would that require replacing `urllib` with `Requests` and rewriting `rest_client.py` and it's subclasses?

**Comment by epicfaace** [comment]
Yeah, I guess that might be beyond the scope of this PR. We could shelve that as another issue and continue ahead with this approach for now.

**Review by yiboliu — APPROVED** [review]
LGTM

File: codalab/common.py [hunk 1]
```diff
@@ -116,3 +120,16 @@ def ensure_str(response):
         return response.decode()
     except UnicodeDecodeError:
         return BINARY_PLACEHOLDER
+
+
+@retry(urllib.error.URLError, tries=2, delay=1, backoff=2)
+def urlopen_with_retry(request: urllib.request.Request, timeout: int = URLOPEN_TIMEOUT_SECONDS):
+    """
+    Makes a request using urlopen with a timeout of 60 seconds and retries on failures.
```
Comment (epicfaace) [inline codalab/common.py]:
```suggestion
    Makes a request using urlopen with a timeout of URLOPEN_TIMEOUT_SECONDS seconds and retries on failures.
```

**Review by epicfaace — APPROVED** [review]

**Pull request merged**